### PR TITLE
Task Instance Modal UX Enhancements

### PR DESCRIPTION
**Pull request by ryanahamilton**
Attempting to make one of the more complex UI interactions more intuitive with the following edits:

### Task Instance Modal

- Adds "X" button to close modal in the top-right as intended by Bootstrap patterns ("Close" button in the bottom-right still persists as well).
- Cleaned up the presentation the modal header by adding "Task Instance/at" label/prefixes and moving the very randomly placed Filter (icon) button down with the options below as a "Filter Upstream" button. 
- Removed the "primary" button styling from each of the buttons in the group row ("Task Instance Details", "Rendered", etc.). This wasn't appropriate usage of a "primary" styling as that designation is used to establish a visual hierarchy—in this case all of the actions are considered comparable.
- Re-ordered each "action" form. (buttons on right instead of left) Forms should flow top to bottom OR left to right. This aligns with the user flow of electing the related options and then submitting the form.
- Improved the accessibility of the "action" forms by refactoring the underlying markup/scripting to each be wrapped in their own `<form></form>` tags so that they can be individually submitted via keyboard (depending on focus).
- Updated the screenshot of this modal [in the docs](https://airflow.apache.org/docs/stable/ui.html#task-instance-context-menu).

| Before | After |
|---|---|
| image  |  image |

### DAG Modal
Applies patterns consistent with updates to the Task Instance Modal:
- Added "DAG:" prefix to the modal header
- Added "X" close button in top-right of modal

| Before | After |
|---|---|
| image|  image |

### CODE REVIEW

**Comment by ashb** [comment]
> Removed the "primary" button styling from each of the buttons in the group row ("Task Instance Details", "Rendered", etc.). This wasn't appropriate usage of a "primary" styling as that designation is used to establish a visual hierarchy—in this case all of the actions are considered comparable.

This is the only change I'm not sure of, as the "primary" styling was used to differentiate toggles from actions, and now it's not visually obvious that the "Task Instance Details" button is an action, but the "Ignore All Deps" is a toggle.

**Comment by ryanahamilton** [comment]
@ashb in a view of 27 buttons visual differentiation is certainly going to be a difficult task regardless of the solution. The attached formation of the Radio Button Groups are meant to differentiate their intended use from the detached/spaced presentation of the row at the top. I'm thinking we might be able to more explicitly separate the user concerns by adding a "Task Actions" heading as a demarcation between them. WDYT? 

image

File: airflow/www/templates/airflow/dag.html [hunk 1]
```diff
@@ -108,199 +108,221 @@ <h4 class="pull-right">
...
-        tabindex="-1" role="dialog"
-      aria-labelledby="dagModalLabel" aria-hidden="true">
+  <div class="modal fade" id="dagModal" tabindex="-1" role="dialog" aria-labelledby="dagModalLabel" aria-hidden="true">
     <div class="modal-dialog">
       <div class="modal-content">
         <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
           <h4 class="modal-title" id="dagModalLabel">
-            <span id='dag_id'></span>
+            <span class="text-muted">DAG:</span> {{dag.dag_id}}
```
Comment (ryanahamilton) [inline airflow/www/templates/airflow/dag.html]: I need to make the "DAG" text dynamic as this modal can be used for "SUBDAG" too.

**Comment by turbaszek** [comment]
> image

I like it 👌

File: airflow/www/templates/airflow/dag.html [hunk 2]
```diff
@@ -108,199 +108,221 @@ <h4 class="pull-right">
...
               <ul class="nav nav-pills" role="tablist" id="redir_log_try_index" style="display:inline">
               </ul>
               <hr/>
               <hr/>
-            {% endif %}
-          </div>
-          <form method="POST">
+            </div>
+          {% endif %}
+          <form method="POST" data-action="{{ url_for('Airflow.run') }}">
```
Comment (ashb) [inline airflow/www/templates/airflow/dag.html:162]: Why `data-action` and not just `action`?

**Comment by ashb** [comment]
Rebase on to latest master to fix test failures please

File: airflow/www/templates/airflow/dag.html (hunk 2 above)
Comment (ryanahamilton) [reply airflow/www/templates/airflow/dag.html:162]: The associated scripting below (line 555) is modifying form values priory to submitting, so it's not submitting to the `action` natively. The second to last line is basically changing `data-action` to `action`—this just felt like an explicit indicator when reading the code that this form is handled by scripting (not a vanilla `<form>`).
```js
    // Task Instance Modal actions
    $('form[data-action]').submit(function (e) {
      e.preventDefault();
      var form = $(this).get(0);
      form.execution_date.value = execution_date;
      form.origin.value = window.location;
      if (form.task_id) {
        form.task_id.value = task_id;
      }
      form.action = $(this).data("action");
      form.submit();
    })
```

**Review by ashb — APPROVED** [review]

**Comment by himabindu07** [comment]
I have verified with this image quay.io/astronomer/ap-airflow-dev:2.0.0-buster-onbuild-22919

**Pull request merged**